lib/caesar.rb

```diff
@@ -1,6 +1,44 @@
 class Caesar
-  attr_reader :key
-
+  attr_reader :character_hash
   def initialize
+    @characters = ("a".."z")
+    @character_hash = {}
+  end
+
+  def rotate_characters(rotation)
+    @characters.to_a.rotate(rotation)
+  end
+
+  def cypher(rotation)
+    rotated = rotate_characters(- rotation)
+    @character_hash = Hash[@characters.zip(rotated)]
+    @character_hash[" "] = " "
```

Comment:
Think about how we can make this work for any nonalphabetic character

Reply:
Thanks for all the comments and insight, really appreciate you spending the time to provide thorough feedback.